### PR TITLE
tests: make integration testing faster

### LINKED ISSUE #690: Understand why it takes so long to pass the tests
Now it takes a very long time to pass the tests (> 1 hour).
What's needed:
* to understand why
* suggest solutions

### PR DESCRIPTION
**Pull request by better0fdead**
Separated integration tests from linter, unit, e2e and examples checks.
Result: ~15% faster testing.
Added pre-building of slow projects.
Total result: ~40-65% faster.
Closes #690

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build mage
```
Comment (LeonidVas) [inline magefile.go:1]: Since you are adding `//go:build mage` remove `//+build mage`.

File: magefile.go [hunk 2]
```diff
@@ -33,6 +34,8 @@ var generatedModeFile = "create_cartrdige_template_filemodes_gen.go"
 
 var completionPath = "./completion"
 
+var build_projects_path = "test/projects"
```
Comment (LeonidVas) [inline magefile.go:37]: Can we create an arbitrary temporary directory and pass it to the tests?

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi! Thank you for the patchset.
Generally I'm OK.
Please add to the commit body the reason and what profit we have.

**Comment by LeonidVas** [comment]
Am I correct in assuming that further ticket work is planned or more ideas to accelerate testing.

File: magefile.go (hunk 1 above)
Comment (better0fdead) [reply magefile.go:1]: Fixed

File: magefile.go (hunk 2 above)
Comment (better0fdead) [reply magefile.go]: Done

File: magefile.go [hunk 3]
```diff
@@ -163,7 +166,23 @@ func Integration() error {
...
-	return sh.RunV(py3Exe, "-m", "pytest", "test/integration")
+	err := sh.RunV(py3Exe, "-m", "pytest", "test/make_preparations/test_build_projects.py")
+	if err != nil {
+		return err
+	}
+	err = sh.RunV(py3Exe, "-m", "pytest", "test/integration")
+	if err != nil {
+		return err
+	}
+	err = os.RemoveAll(build_projects_path)
```
[the remark this answers is not in the document]
Comment (better0fdead) [reply magefile.go]: Re-wrote, now it works like you commented.

File: magefile.go [hunk 4]
```diff
@@ -161,9 +161,19 @@ func Unit() error {
 // Run integration tests
 func Integration() error {
 	mg.Deps(GenerateGoCode)
-
+	build_projects_path := os.TempDir() + "/cartridge_cli_test_temp"
+	err := os.Mkdir(build_projects_path, 0777)
```
Comment (LeonidVas) [inline magefile.go:165]: 0777 - looks like overkill (write description in comments).

**Comment by LeonidVas** [comment]
> Am I correct in assuming that further ticket work is planned or more ideas to accelerate testing?

How about this question?

**Comment by better0fdead** [comment]
> > Am I correct in assuming that further ticket work is planned or more ideas to accelerate testing?
> 
> How about this question?

In my opinion, at that state it is more profitable to do that issue #724, or think about  #730. I suggest closing that issue for now. Total profit is ~50% from initial point.

**Review by LeonidVas — APPROVED** [review]
LGTM

**Pull request merged**